crates/proof-of-sql/src/sql/proof_gadgets/shift.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{
+    base::{
+        database::Column, polynomial::MultilinearExtension, proof::ProofError, scalar::Scalar,
+        slice_ops,
+    },
+    sql::proof::{
+        FinalRoundBuilder, FirstRoundBuilder, SumcheckSubpolynomialType, VerificationBuilder,
+    },
+};
+use alloc::{boxed::Box, vec};
+use bumpalo::Bump;
+use num_traits::One;
+
+/// Perform first round evaluation of downward shift.
+pub(crate) fn first_round_evaluate_shift<'a, S: Scalar>(
+    builder: &mut FirstRoundBuilder<'a, S>,
+    num_rows: usize,
+) {
+    builder.produce_rho_evaluation_length(num_rows as usize);
+    builder.request_post_result_challenges(1);
```

Comment:
If we have this request within the gadget, it makes sense to consume it within the gadget as well. So, alpha shouldn't be a parameter for the other parts.

Optionally, remove it here (if we want to reuse challenges across gadgets).